### DISCUSSION TITLE
Use of tags in West manifest

### DISCUSSION THREAD

**Discussion by hoyes** [zephyrproject-rtos/zephyr#55780] [General]
Over at the [meta-zephyr Yocto layer](https://git.yoctoproject.org/meta-zephyr) we have a [script](https://git.yoctoproject.org/meta-zephyr/tree/meta-zephyr-core/scripts/generate-version.py) which we use to automatically generate Bitbake configuration for a specific Zephyr release (using West as a library).

Yocto/OE [advises](https://docs.yoctoproject.org/ref-manual/variables.html?highlight=srcrev#term-SRCREV) setting the SRCREV for each module to a full SHA, not a tag, the idea being that tags can change but SHAs are immutable.

I notice that a [tag was recently accepted into the West manifest](https://github.com/zephyrproject-rtos/zephyr/pull/51219#discussion_r998123145) for zcbor - I wondered if someone could clarify Zephyr policy on this, i.e.:
 * Should I raise a PR here to change the tag to a SHA, or
 * Update our meta-zephyr script to resolve tags to SHAs where necessary.

Thanks in advance for your assistance,

Peter

**Comment by yashi** [comment]
I'm not a maintainer of Zephyr nor Yocto project but using either a tag or a hash value in a West manifest doesn't mean anything to Yocto project.  Technically a tag in Git _can_ change but that's highly discouraged, especially for public tags.

> Should I raise a PR here to change the tag to a SHA

Please don't.  We are free to choose whatever suites us. I mean you can send a PR, but I don't see any benefit for Zephyr project.  Moreover, what if a Zephyr module is from outside of Zephyr project and a tag is used in a West manifest?  We, as in Zephyr project, can't ensure the tag is always resolve to exact hash value.

> Update our meta-zephyr script to resolve tags to SHAs where necessary.

If this means make your scripts to resolve a tag automatically, then, it's up to you.  In fact, as you already know, BitBake allows you to use a tag.  It just need to access network to resolve it, I suppose.  And a part of Yocto users don't like network access while building it. (I can understand the requirement)

Or, if this means you personally run `git rev-parse <tag>` and paste the hash value in your layer, that's fine as well.  I know I can't promiss it or my word doesn't mean anything, but I don't think Zephyr project will change the tags we already have in public.

All in all, if you really don't want to use tags, have a local west manifest without tags?

My two cents.
